scripts/ci/processCoverageReport.py

```diff
@@ -0,0 +1,414 @@
+            print(f"Environment variable '{name}' is set")
+        else:
+            print(f"Environment variable '{name}' = '{value}'")
+    else:
+        print(f"Environment variable '{name}' is not set, using default: '{default}'")
+
+    return value
+
+def parse_jacoco_xml(xml_path):
+    """Parse a JaCoCo XML report and extract coverage data."""
+    try:
+        print(f"Parsing JaCoCo XML file: {xml_path}")
+        tree = ET.parse(xml_path)
+        root = tree.getroot()
+
+        overall_coverage = {}
+        for counter in root.findall(".//counter"):
+            type_name = counter.get('type')
+            covered = int(counter.get('covered', 0))
+            missed = int(counter.get('missed', 0))
+
+            overall_coverage[type_name] = {
+                'covered': covered,
+                'missed': missed,
+                'total': covered + missed,
+                'percentage': (covered / (covered + missed) * 100) if (covered + missed) > 0 else 0
+            }
+
+        file_coverage = {}
+
+        for package in root.findall(".//package"):
+            package_name = package.get('name', '').replace('/', '.')
+
+            for cls in package.findall("./class"):
+                class_name = cls.get('name', '')
+                source_file = cls.get('sourcefilename', '')
+
+                if not source_file:
+                    continue
+
+                file_path = f"{package_name.replace('.', '/')}/{source_file}"
+
+                if file_path not in file_coverage:
+                    file_coverage[file_path] = {
+                        'package': package_name,
+                        'source_file': source_file,
+                        'full_path': file_path,
+                        'counters': defaultdict(lambda: {'covered': 0, 'missed': 0})
+                    }
+
+                for counter in cls.findall("./counter"):
+                    type_name = counter.get('type')
+                    covered = int(counter.get('covered', 0))
+                    missed = int(counter.get('missed', 0))
+
+                    file_coverage[file_path]['counters'][type_name]['covered'] += covered
+                    file_coverage[file_path]['counters'][type_name]['missed'] += missed
+
+        for file_path, file_data in file_coverage.items():
+            for counter_type, counter in file_data['counters'].items():
+                total = counter['covered'] + counter['missed']
+                counter['total'] = total
+                counter['percentage'] = (counter['covered'] / total * 100) if total > 0 else 0
+
+        print(f"Successfully parsed {xml_path} - Found {len(file_coverage)} files with coverage data")
+        return {
+            'overall': overall_coverage,
+            'files': file_coverage
+        }
+
+    except Exception as e:
+        print(f"Error parsing JaCoCo XML file {xml_path}: {str(e)}")
+        return {'overall': {}, 'files': {}}
+
+def get_git_changed_files():
```

Comment:
We already have a GHA that does this for the PMD scan. Can we reuse anything from that?

Reply:
Ok - sounds good.